lib/atc/gcp/storage_uploader.rb

```diff
@@ -0,0 +1,76 @@
+# frozen_string_literal: true
+
+# rubocop:disable Rails/Output
+
+class Atc::Gcp::StorageUploader
+  def initialize(storage_client, bucket_name)
+    @storage_client = storage_client
+    @bucket_name = bucket_name
+  end
+
+  # Uploads a file to a Google Cloud Storage
+  # @param local_file_path [String] Path to a local file to upload.
+  # @param object_key [String] Bucket key to use for the uploaded object.
+  # @param options [Hash] A hash of options.
+  # @option options [Boolean] :overwrite (boolean)
+  #   If an object already exists at the target object_key, { overwrite: true } will overwrite it
+  #   and { overwrite: false } will raise an exception.
+  #   Default value: false
+  # @option options [String] :precalculated_whole_file_crc32c
+  #   If this option is provided, it will be send to Google during transfer and Google will verify that
+  #   the content received matches this checksum.  If this option is omitted, a local crc32c checksum
+  #   will be generated right before the upload occurs, and that checksum will be verified with Google.
+  # @option options [Boolean] :verbose
+  #   If provided, this will print progress messages to stdout (via puts).
+  #   Default value: false
+  # @option options [Hash] :metadata
+  #   A Hash of key-value metadata pairs (and combined length of all keys and values cannot be
+  #   greater than 8192 bytes).
+  # @return True if the upload succeeded, or throws Atc::Exceptions::TransferError if the transfer failed.
+  def upload_file(local_file_path, object_key, **options)
```

Comment:
This is a common signature with the S3 uploader, which is 👍  - but I wonder if we should pull metadata out as a keyword arg in a post-merge commit.

Reply:
Good thought! 👍